docs/features/logforwarding/outputs/splunk-forwarding.adoc

```diff
@@ -43,5 +49,135 @@ spec:
+* Only *string values* are allowed.
+
+==== Extended Support via Remap
+
+To provide more flexibility, we extend the functionality with a *custom remap function*.
+This allows you to *work around some of these limitations* by transforming field names and values before forwarding.
+
+With this remapping:
+
+* *Nested fields* can be flattened into top-level fields.
+* Field *paths* are flattened using *dot notation*, and unsupported characters are replaced with underscores (`_`).
+* *Non-string values* are automatically converted to strings.
+* *Objects* are serialized as JSON strings.
+
+==== Examples
+
+[source,json]
+----
+{
+  "annotations": {
+    "authorization.k8s.io/decision": "allow",
+    "authorization.k8s.io/reason": "foo_bar"
+  }
+}
+----
+
+Becomes:
+
+[source,json]
+----
+{
+  "annotations_authorization_k8s_io_decision": "allow",
+  "annotations_authorization_k8s_io_reason": "foo_bar"
+}
+----
+
+Non-string values:
+
+[source,json]
+----
+{
+  "retries": 3,
+  "success": true
+}
+----
+
+Becomes:
+
+[source,json]
+----
+{
+  "retries": "3",
+  "success": "true"
+}
+----
+
+Nested object values:
+
+[source,json]
+----
+{
+  "objectRef": {
+    "resource": {
+      "status": 200,
+      "statusText": "OK"
+    }
+  }
+}
+----
+
+Becomes:
+
+[source,json]
+----
+{
+  "objectRef_resource": "{\"status\":200,\"statusText\":\"OK\"}"
+}
+----
+[IMPORTANT]
+====
+After remapping, the *original fields* will be removed to avoid duplication.
+Only the transformed, top-level fields will be retained in the final log event.
+====
+
+
+=== `sourceType` and `host`
+
+The `sourceType` and `host` fields are *not configurable* through the Cluster Log Forwarder API. They are set automatically as described below, independent of the `.log_type` or `.log_source` values.
+
+* `sourceType`: Set to `_json`
+This reflects the fact that all log events are forwarded as structured JSON objects.
+
+* `host`: Set to the value of `.hostname`
+This ensures that each log event carries the correct *originating host* information.
+
+== Default settings
+Below the table with default value depends on log_type and log_source will be used if not set in configuration.
+
+|===
+| |pass:[Infrastructure Journal <br/><em>log_type:infrastructure<br/>log_source:node</em>]|pass:[Infrastructure/Application Container<br/><em>log_type:infrastructure\|application<br/>log_source:container</em>]|pass:[Audit<br/><em>log_type:audit<br/>log_source:auditd\|ovn\|openshiftAPI\|kubeAPI<em/>]|Note
+|`index`|||| not configured by default
+|`source`|SYSLOG_IDENTIFIER|ns_name_podName_containerName|.log_source|
+|`indexedFields`|||| not configured by default
+|`sourceType`|`_json`|`_json`|`_json`| *not configurable*
+|`host`|`.hostname`|`.hostname`|`.hostname`|*not configurable*
 
-NOTE: This will forward logs to the log type of the message.  The default index of the splunk server configuration is used when 'index' is not defined
+|===
```

Comment:
Let's add 'message' to this table and state that it is the "ViaQ envelope which is a JSON string".  Please confirm this is currently what we send

Reply:
@jcantrill you mean `event`?

Reply:
My original statement I meant message but given this is for splunk and that is how users of splunk will associate the information in the splunk API, I agree 'event' is a better choice.  Since we talked about allowing this to be configurable then maybe we should also call it "EventKey" in our API instead of "PayloadKey" so the words are consistent with splunk's API.  @cahartma thoughts?  Does it make more sense to make our API consistent (i.e. Payload Key is what we use in syslog) or make it consistent with the sink (i.e. splunk users know it as 'event'0?